### PR TITLE
Enhanced plugin config comparison

### PR DESCRIPTION
**Pull request by shaynevanasperen**

### CODE REVIEW

File: test/Kongverge.IntegrationTests/ProgramSteps.cs [hunk 1]
```diff
@@ -116,28 +117,40 @@ protected async Task OutputFolderContentsMatchesFolderContentsOf(string name)
...
                     var folderServiceRoute = folderService.Routes.Single(x => x.Equals(outputServiceRoute));
+                    await FillInDefaults(folderServiceRoute.Plugins, availablePlugins);
                     outputServiceRoute.Plugins.Should().BeEquivalentTo(folderServiceRoute.Plugins);
                 }
             }
         }
+
+        private static async Task FillInDefaults(IEnumerable<KongPlugin> plugins, IDictionary<string, AsyncLazy<KongPluginSchema>> availablePlugins)
+        {
+            foreach (var globalConfigPlugin in plugins)
```
Comment (TomFreeman) [inline test/Kongverge.IntegrationTests/ProgramSteps.cs]: The name globalConfigPlugin is misleading isn't it?
This function is applied to all objects with plugins.

File: src/Kongverge/DTOs/KongPlugin.cs [hunk 2]
```diff
@@ -73,5 +75,45 @@ public virtual Task Validate(IReadOnlyCollection<string> availablePlugins, IColl
...
+                Config = new Dictionary<string, object>();
+            }
+            FillInField(schema, Config);
+        }
+
+        private static void FillInField(KongPluginSchema schema, IDictionary<string, object> config)
+        {
+            foreach (var fieldDefinition in schema.Fields)
+            {
+                if (config.ContainsKey(fieldDefinition.Key))
```
Comment (TomFreeman) [inline src/Kongverge/DTOs/KongPlugin.cs:89]: This section is a took a little parsing.

Would be nice to either wrap the bodies of the clause with descriptive method names (e.g. `CreateDefaultObject` and `FillInBlankDefaults` as examples) or just to have similar comments.)

**Review by TomFreeman — DISMISSED** [review]
Nit picks for future maintainers, no blocking issues.

File: test/Kongverge.IntegrationTests/ProgramSteps.cs (hunk 1 above)
Comment (shaynevanasperen) [reply test/Kongverge.IntegrationTests/ProgramSteps.cs]: Ah, yes, that was leftover from extracting a method from where it was using just the global plugins. Will fix.

File: src/Kongverge/DTOs/KongPlugin.cs (hunk 2 above)
Comment (shaynevanasperen) [reply src/Kongverge/DTOs/KongPlugin.cs:89]: Will also add some unit test coverage for this now. It already has coverage by integration tests but that's not explicitly checking this behaviour.

**Review by TomFreeman — APPROVED** [review]

**Pull request merged**